### PR TITLE
Towards CRAN (second batch)

### LINKED ISSUE #139: rhdf5lib version issues
how do we resolve the ```Rhdf5lib``` issue? Using the newest version of ```rhdf5``` requires it, but that in turn requires ```R >= 3.5```. Should we just depend on an older version of ```rhdf5``` for the time being? Without a workaround, ```R CMD check``` fails right-away.

### PR DESCRIPTION
**Pull request by martin-ueding**
This is the second batch for Issue #95.

Resolves #139.

### CODE REVIEW

**Comment by kostrzewa** [comment]
@urbach @martin-ueding While the effect on testability and general documentation coverage of this effort is great, do you feel that the time is well spent? There's a lot of work still left to do if I understand correctly...

**Comment by urbach** [comment]
it should clearly not be highest priority!

**Comment by kostrzewa** [comment]
> it should clearly not be highest priority!

The point is that if we want this, then we need to do it sooner rather than later because the code is growing and changing now. Keeping it CRAN-compatible will be a burden (although one which might be worth it), but getting it CRAN-compatible now is easier than in a few months' time where we might have lots of new stuff in the codebase...

**Comment by urbach** [comment]
functions `averx`, `pion_ff`, `summary.pionff`, `summary.averx`, `print.averx` could be moved to the `exec` directory.

**Comment by urbach** [comment]
`Rhdf5lib` not available for my R-version (3.4.4) :(

**Comment by martin-ueding** [comment]
Regarding the versions it is a bit messy:

| R | Bioc | rhdf5 | Rhdf5lib dependency | Rhdf5lib |
| --- | --- | --- | --- | --- |
| 3.4 | 3.6 | [2.22.0](https://bioconductor.org/packages/3.6/bioc/html/rhdf5.html) | None | [1.0.0](https://bioconductor.org/packages/3.7/bioc/html/Rhdf5lib.html) |
| 3.5 | 3.7 | [2.24.0](https://bioconductor.org/packages/3.7/bioc/html/rhdf5.html) | Unversioned | [1.2.1](https://bioconductor.org/packages/3.7/bioc/html/Rhdf5lib.html) |
| 3.5 | 3.8 | [2.26.2](https://bioconductor.org/packages/3.8/bioc/html/rhdf5.html) | ≥ 1.3.2 | [1.4.2](https://bioconductor.org/packages/3.8/bioc/html/Rhdf5lib.html) |

We depend on `release`, that currently is Bioc 3.8 and therefore needs R 3.5. In principle do not depend on a particular version of Bioc. For some reason the `Rhdf5lib` dependency does not get resolved properly when we only depend on `rhdf5`. It pulls down a version smaller than `1.3.2` on my system and that does not work with my system installation of HDF5. On your Linux Mint I suppose that HDF5 is also a bit older and that Bioc 3.6 should work then.

Let me try to remove the explicit version dependency and see whether it still works on Travis CI.

**Comment by martin-ueding** [comment]
Even without the explicit version it installs the correct ones:

```
 rhdf5         2.26.2   2019-01-08 [1] bioc_git2r (@81e1125)
 Rhdf5lib      1.4.2    2019-01-08 [1] bioc_git2r (@86b4372)
```

As seen in the table from my last comment the `rhdf5` version matching R 3.4 does not depend on `Rhdf5lib`. So let me try to remove this from the dependencies.

**Comment by martin-ueding** [comment]
I do not understand why I had to include `Rhdf5lib`, but it seems to work just fine without now. If we are lucky, then `release` means 3.6 for you and it should now work in your environment.

**Comment by urbach** [comment]
> I do not understand why I had to include `Rhdf5lib`, but it seems to work just fine without now. If we are lucky, then `release` means 3.6 for you and it should now work in your environment.

Now it works for me...

File: R/boot_ts_array.R [hunk 1]
```diff
@@ -4,6 +4,32 @@
 #' on it. It is bad to depend on private functions of a library, so we copy it
```
Comment (urbach) [inline R/boot_ts_array.R]: So, I think for the time being we go for this copy of the function

File: R/jackknifeafterboot.R [hunk 2]
```diff
@@ -5,26 +5,26 @@ jab <- function(t, t0, starts, m=1, fn=sd) {
   
   jack.boot <- function(indices, xstar, f) {
```
Comment (urbach) [inline R/jackknifeafterboot.R]: here I have to check whether or not this function really does what it should, but this is not related to the current pull request

File: man/effectivemass.cf.Rd [hunk 3]
```diff
@@ -4,7 +4,7 @@
 \description{
   Computes effective mass values for a correlation function using
   different type of definitions for the effective mass. This function is
-  mainly indented for internal usage by \code{\link{bootstrap.effectivmass}}.
+  mainly indented for internal usage by \code{\link{bootstrap.effectivemass}}.
```
Comment (urbach) [inline man/effectivemass.cf.Rd]: so, mainly indented for internal usage? Then, do not export...

**Comment by martin-ueding** [comment]
> Maybe we can generate a list of functions with missing docu?

Of course, that is generated as part of `R CMD check`:

```
Undocumented code objects:
  ‘analysis_gradient_flow’ ‘analysis_online’ ‘avg.cbt.cf’ ‘avg.ls.cf’
  ‘bootstrap.hankel’ ‘bootstrap.meanerror’ ‘compute.plotlims’
  ‘compute.qtildesq’ ‘compute.qtildesq.contdisp’ ‘correlatormatrix’
  ‘effmass2’ ‘escapeLatexSpecials’ ‘g1’ ‘getorderedconfignumbers’
  ‘gevp.hankel’ ‘gevp.hankel.old’ ‘invcosh’ ‘jab.cf.derived’
  ‘jab.effectivemass’ ‘jab.effectivemassfit’ ‘jab.matrixfit’
  ‘jackknife_cov’ ‘pcacfit’ ‘plot_eigenvalue_timeseries’
  ‘plot_timeseries’ ‘plot.b1fit’ ‘plot.cfit’ ‘plot.effectivemass’
  ‘plot.effmass’ ‘plot.gevp.amplitude’ ‘plot.hadronacf’ ‘plot.massfit’
  ‘plot.ofit’ ‘plot.pionff’ ‘plot.pionfit’ ‘plot.rhofit’
  ‘plothlinewitherror’ ‘print.averx’ ‘print.cf’
  ‘print.effectivemassfit’ ‘print.ofit’ ‘proton’ ‘pscor.sample’
  ‘readhlcor’ ‘summary.averx’ ‘summary.cf’ ‘summary.effectivemass’
  ‘summary.effectivemassfit’ ‘summary.gevp.amplitude’
  ‘summary.hadronacf’ ‘summary.matrixfit’ ‘summary.ofit’
  ‘summary.pionff’ ‘summary.uwerr’ ‘symmetrise.cf’ ‘takeTimeDiff.cf’
  ‘uwerr.cf’ ‘weighted.hist’ ‘weighted.median1’ ‘weighted.quantile’
  ‘weighted.variance’
Undocumented data sets:
  ‘correlatormatrix’ ‘pscor.sample’
```

> Moreover, we should generate a list of functions that not to be exportet!

We can just remove them from the `NAMESPACE` file in this branch I guess.

**Comment by urbach** [comment]
> > Maybe we can generate a list of functions with missing docu?
> 
> Of course, that is generated as part of `R CMD check`:
> 
> ```
> Undocumented code objects:
>   ‘analysis_gradient_flow’ ‘analysis_online’ ‘avg.cbt.cf’ ‘avg.ls.cf’
>   ‘bootstrap.hankel’ ‘bootstrap.meanerror’ ‘compute.plotlims’
>   ‘compute.qtildesq’ ‘compute.qtildesq.contdisp’ ‘correlatormatrix’
>   ‘effmass2’ ‘escapeLatexSpecials’ ‘g1’ ‘getorderedconfignumbers’
>   ‘gevp.hankel’ ‘gevp.hankel.old’ ‘invcosh’ ‘jab.cf.derived’
>   ‘jab.effectivemass’ ‘jab.effectivemassfit’ ‘jab.matrixfit’
>   ‘jackknife_cov’ ‘pcacfit’ ‘plot_eigenvalue_timeseries’
>   ‘plot_timeseries’ ‘plot.b1fit’ ‘plot.cfit’ ‘plot.effectivemass’
>   ‘plot.effmass’ ‘plot.gevp.amplitude’ ‘plot.hadronacf’ ‘plot.massfit’
>   ‘plot.ofit’ ‘plot.pionff’ ‘plot.pionfit’ ‘plot.rhofit’
>   ‘plothlinewitherror’ ‘print.averx’ ‘print.cf’
>   ‘print.effectivemassfit’ ‘print.ofit’ ‘proton’ ‘pscor.sample’
>   ‘readhlcor’ ‘summary.averx’ ‘summary.cf’ ‘summary.effectivemass’
>   ‘summary.effectivemassfit’ ‘summary.gevp.amplitude’
>   ‘summary.hadronacf’ ‘summary.matrixfit’ ‘summary.ofit’
>   ‘summary.pionff’ ‘summary.uwerr’ ‘symmetrise.cf’ ‘takeTimeDiff.cf’
>   ‘uwerr.cf’ ‘weighted.hist’ ‘weighted.median1’ ‘weighted.quantile’
>   ‘weighted.variance’
> Undocumented data sets:
>   ‘correlatormatrix’ ‘pscor.sample’
> ```

so, I think we do not need

effmass2
gevp.hankel.old

and we do not need to export

g1
plothlinewitherror
print.ofit

plot.cfit is directly called by plot.pionfit, plot.rhofit, plot.b1fit,
plot.ofit, so they are just the same.

I' not export

‘weighted.hist’ ‘weighted.median1’ ‘weighted.quantile’                                                                                                                                                                         
   ‘weighted.variance’

comments?

**Comment by urbach** [comment]
I'd like to remove `weightedstuff.R` and suggest the usage of the `spatstat` package (where I took them from in the old days...) What do you think?

**Comment by urbach** [comment]
`weighted.hist` needs another solution...

**Comment by kostrzewa** [comment]
> I'd like to remove weightedstuff.R and suggest the usage of the spatstat package (where I took them from in the old days...) What do you think?

Sure, that's fine. I did actively use ```weightedstuff.R``` in various analyses. I did not use ```weightet.hist``` as far as I recall, using instead a function from some package which I don't remember now.

**Comment by kostrzewa** [comment]
> > I'd like to remove weightedstuff.R and suggest the usage of the spatstat package (where I took them from in the old days...) What do you think?
> 
> Sure, that's fine. I did actively use `weightedstuff.R` in various analyses. I did not use `weightet.hist` as far as I recall, using instead a function from some package which I don't remember now.

I used the function from the ```plotrix``` package as this was more flexible for reasons which I don't remember.

**Comment by urbach** [comment]
> > > I'd like to remove weightedstuff.R and suggest the usage of the spatstat package (where I took them from in the old days...) What do you think?
> > 
> > Sure, that's fine. I did actively use `weightedstuff.R` in various analyses. I did not use `weightet.hist` as far as I recall, using instead a function from some package which I don't remember now.
> 
> I used the function from the ```plotrix``` package as this was more flexible for reasons which I don't remember.

I think I took it from there. So, I'll remove that file then!?

**Comment by kostrzewa** [comment]
> > > I'd like to remove weightedstuff.R and suggest the usage of the spatstat package (where I took them from in the old days...) What do you think? > > Sure, that's fine. I did actively use `weightedstuff.R` in various analyses. I did not use `weightet.hist` as far as I recall, using instead a function from some package which I don't remember now. I used the function from the ```plotrix``` package as this was more flexible for reasons which I don't remember.
> I think I took it from there. So, I'll remove that file then!?

Yes, indeed it's the same funciton. So perhaps I used one from yet another package, but it's unimportant, the ```plotrix``` one does just fine as a drop-in replacement.

**Comment by martin-ueding** [comment]
Do we want to merge this batch of changes such that the `Rhdf5lib` dependency is cleaned up in `master`?

**Comment by martin-ueding** [comment]
Resolves #139.

**Comment by martin-ueding** [comment]
As Carsten said that this could be merged in the seminar, I take that as an approval.

**Pull request merged**